### ISSUE TITLE
New extension: access log based metrics

### ISSUE THREAD

**Issue by kyessenov** [envoyproxy/envoy#30619]
This is a proposal to add a new access logger that emits metrics using the Envoy metrics system. An example instance of this extension would be [Istio standard metrics](https://istio.io/latest/docs/reference/config/metrics/) that are derived from the access log entries inside the proxy.

Note that the concern about the scalability does apply, because an access log value cardinality is unbounded. A few mitigation measures must be placed on this extension to be safe to use in production:
* the metrics scopes must have a limited lifetime;
* integration with the overload manager to drop the metrics with the excessive cardinality;

CC @jmarantz @briansonnenberg 

CC #29681

**Comment by wbpcode** [comment]
But how we can manage the lifetime of these metrics? Seems these metrics may have a global or listener scope?

**Comment by kyessenov** [comment]
Yeah, they can be scoped to the listener or global. The lifetime for metrics needs to be decoupled from xDS parent resources because the cardinality depends on the traffic. That means the scope should be rotated periodically without the parent reload, for example.

**Comment by kyessenov** [comment]
https://github.com/open-telemetry/opentelemetry-specification/blob/7322f8bad298f1fbe872e1a4e1462aee931196d6/specification/metrics/sdk.md#cardinality-limits is relevant since it has reasonable caps on to prevent metric overflow.

**Comment by kyessenov** [comment]
We could start with an opinionated set of base metrics and labels per https://opentelemetry.io/docs/specs/semconv/http/http-metrics/.

**Comment by kyessenov** [comment]
Fixed by https://github.com/envoyproxy/envoy/pull/41770.
